project-explorer.el

```diff
@@ -1736,6 +1742,45 @@ outside of the project's root."
                                       'refresh
                                     'directory-change)))))))
 
+;; Follow the currently opened buffer and show the corresponding
+;; project
+(defvar pe/follow-current-timer nil
+  "Timer to limit the frequency of the current project check.")
+
+(defun pe/has-cache ()
+  "Check if the current buffer has a cached project associated
+with it."
+  (when pe/cache-enabled
+    (let ((dir (funcall pe/project-root-function)))
+      (and dir (file-exists-p (pe/cache-make-filename dir))))))
+
+(defun pe/follow-current-open ()
+  "Switch the project staying in the same window.
+If project-explorer isn't opened or no project associated with
+the current buffer, stop following."
+  (if (pe/get-project-explorer-window)
+      (when (or (pe/get-current-project-explorer-buffer)
+                (pe/has-cache))
+        (let ((win (selected-window)))
+          (project-explorer-open)
+          (select-window win)))
+    (pe/stop-follow-current)))
```

Comment:
You probably want to check `pe/follow-current`, and not `(pe/get-project-explorer-window)` when disabling. What if a user has multiple frames?